src/Http/Client.php

```diff
@@ -479,7 +488,9 @@ public function send(RequestInterface $request, array $options = []): Response
         }
 
         do {
+            $this->dispatchEvent('HttpClient.beforeSend', [$request, $options]);
```

Comment:
Should we check the event for a result? If we want to accomodate the caching/rate limiting use cases that @asgraf raised, we would need to check the `beforeSend` event for a `Response` result as a listener could set a response to bypass sending a request.

Reply:
I think we should have behavioral consistency with how other parts of the framework handle `before` events.